packages/create-react-app/createReactApp.js

```diff
@@ -606,6 +618,67 @@ function isSafeToCreateProjectIn(root, name) {
   return false;
 }
 
+function checkThatNpmCanReadCwd() {
+  const cwd = process.cwd();
+  let childOutput = null;
+  try {
+    // Note: intentionally using spawn over exec since
+    // the problem doesn't reproduce otherwise.
+    // `npm config list` is the only reliably way I could find
+    // to reproduce the wrong path. Just printing process.cwd()
+    // in a Node process was not enough.
+    childOutput = spawn.sync('npm', ['config', 'list']).output.join('');
+  } catch (err) {
+    // Something went wrong spawning node.
+    // Not great, but it means we can't do this check.
+    // We might fail later on, but let's continue.
+    return true;
+  }
+  if (typeof childOutput !== 'string') {
+    return true;
+  }
+  const lines = childOutput.split('\n');
+  // `npm config list` output includes the following line:
+  // "; cwd = C:\path\to\current\dir" (unquoted)
+  // I couldn't find an easier way to get it.
+  const prefix = '; cwd = ';
+  const line = lines.find(line => line.indexOf(prefix) === 0);
+  if (typeof line !== 'string') {
+    // Fail gracefully. They could remove it.
+    return true;
+  }
+  const npmCWD = line.substring(prefix.length);
+  if (npmCWD === cwd) {
```

Comment:
Not sure if this needs to be more relaxed regarding case sensitivity. But we can probably release like this and then wait for issues to come up. I'd prefer it be more strict and relaxed later than vaguely relaxed just in case.

Reply:
Windows and MacOS are both case-insensitive, so I wonder if any strange issues will come up here. In the past I've encountered issues where some places used `c:\` while others used `C:\`.

Does Node.js not have a method to determine if one path is a substring of another??

Reply:
I couldn't find one.